modules/ui/changeset_editor.js

```diff
@@ -74,10 +74,12 @@ export function uiChangesetEditor(context) {
                 osm.userChangesets(function (err, changesets) {
                     if (err) return;
 
+                    var deleted = (JSON.parse(localStorage.getItem('deletedChangesetComments') || '[]')).map(s => s.trim().toLowerCase());
```

Comment:
This logic `.map(s => s.trim().toLowerCase());` is duplicated in a lot of places. Do we really need it? We only ever compare the full string, to it should be OK to compare it as it. Otherwise we might want to extract it into a helper?

Another way would be to only store and compare hashes of those lines… but I don't think that is needed / has benefits.

Reply:
I would even go as far as to store the changeset id of the to be ignored comments instead of the actual literal comment string itself. That would be more memory efficient and be the most stable implementation.

Reply:
what also needs to be taken into account in the store-changeset-id approach is that in case that there exists more than one changeset with the to be ignored comment, all respective changeset ids need to be added to the list. but this does come with an implicit benefit: when the user at some later point manually enters a previously deleted comment, it will be again visible in the suggestions (until deleted again). When storing the comment itself, a user would not be able to undo an accidental deletion of a specific comment.